lib/ecto/repo.ex

```diff
@@ -654,7 +654,8 @@ defmodule Ecto.Repo do
       end)
 
   """
-  @callback transaction(Keyword.t, fun) :: {:ok, any} | {:error, any}
+  @callback transaction(fun | Ecto.Multi.t, Keyword.t) ::
+    {:ok, any} | {:error, any} | {:error, atom, any, %{atom => any}}
```

Comment:
It is fine. Let's just make clear in the docs you get the four item tuple only if a multi is given.